services/src/main/java/org/keycloak/forms/login/freemarker/model/AuthenticationContextBean.java

```diff
@@ -63,7 +63,7 @@ public String getAttemptedUsername() {
         String username = context.getAuthenticationSession().getAuthNote(AbstractUsernameFormAuthenticator.ATTEMPTED_USERNAME);
 
         // Fallback to real username of the user just if attemptedUsername doesn't exist
-        if (username == null) {
+        if (username == null && context.getUser() != null) {
```

Comment:
Only for defensive purposes or did you reach any NPE?

Reply:
I really reached NPE so I added this check